sycl/source/detail/queue_impl.hpp

```diff
@@ -117,11 +117,11 @@ class queue_impl : public std::enable_shared_from_this<queue_impl> {
   /// constructed.
   /// \param AsyncHandler is a SYCL asynchronous exception handler.
   /// \param PropList is a list of properties to use for queue construction.
-  queue_impl(device_impl &Device, const ContextImplPtr &Context,
+  queue_impl(device_impl &Device, std::shared_ptr<context_impl> &&Context,
```

Comment:
Need the rvalue-reference overload for the creation using `getDefaultOrNew` helper at line 108. Maybe this can be refactored further somehow, but that would be independent of passing SYCL RT objects by raw ptr/ref.